packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin.tsx

```diff
@@ -242,7 +242,14 @@ export const TooltipPlugin: React.FC<TooltipPluginProps> = ({
       }
 
       if (sortOrder !== SortOrder.None) {
-        series.sort((a, b) => arrayUtils.sortValues(sortOrder)(a.value, b.value));
+        // create sort reference series array, as Array.sort() mutates the original array
+        const sortRef = [...series];
+        series.sort((a, b) => {
+          // get compared values indices to retrivie raw values from sortIdx
+          const aIdx = sortRef.indexOf(a);
+          const bIdx = sortRef.indexOf(b);
+          return arrayUtils.sortValues(sortOrder)(sortIdx[aIdx], sortIdx[bIdx]);
```

Comment:
can we pull `arrayUtils.sortValues(sortOrder)` up out of this callback?

Reply:
sure, that's fixed now